### PR TITLE
SM-1012: Removing SM Beta (Phase 2)

### PR DESCRIPTION
**Pull request by coltonhurst**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Now that phase 1 of SM beta is complete, in phase 2 we remove all references to SecretsManagerBeta in the server project, but leave the references in the db for phase 3. We need to do this to ensure compatibility between the current and previous versions of server with the database.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by Thomas-Avery — CHANGES_REQUESTED** [review]
Looks good, just need to add EF migrations.

File: src/Core/AdminConsole/Entities/Organization.cs [hunk 1]
```diff
@@ -78,7 +78,6 @@ public class Organization : ITableObject<Guid>, ISubscriber, IStorable, IStorabl
     public int? SmServiceAccounts { get; set; }
     public int? MaxAutoscaleSmSeats { get; set; }
     public int? MaxAutoscaleSmServiceAccounts { get; set; }
-    public bool SecretsManagerBeta { get; set; }
```
[the remark this answers is not in the document]
Comment (coltonhurst) [reply src/Core/AdminConsole/Entities/Organization.cs:81]: Good call, thank you! Fixed in [d6a8a0f](https://github.com/bitwarden/server/pull/3663/commits/d6a8a0facbd715ea4eff3a19af6079b44389fcd1)

**Review by Thomas-Avery — DISMISSED** [review]
Looks good.

**Review by rkac-bw — APPROVED** [review]
looks good to me

**Review by vincentsalucci — APPROVED** [review]
Reviewed the AC team owned files. Quick question: how **good** did making this PR feel? 😄

**Review by cturnbull-bitwarden — APPROVED** [review]

**Comment by coltonhurst** [comment]
@vincentsalucci @cturnbull-bitwarden @rkac-bw @Thomas-Avery I'll need a review one more time if you don't mind!

This has been approved by QA, but I re-ran the EF migrations so that the date would be updated. This should be the only change since the last review.

Thank you!

**Review by vincentsalucci — APPROVED** [review]
Approved AC file changes 👍

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by Thomas-Avery — APPROVED** [review]

**Review by rkac-bw — APPROVED** [review]
Looks good to me

**Pull request merged**